front-end/src/components/Login/index.js

```diff
@@ -40,53 +44,113 @@ class Login extends Component {
     }
   }
 
+  form = React.createRef();
```

Comment:
Is there a good reason to use refs here? The code looks somewhat more complicated now.

Reply:
using refs in react is better idea, because of speed